src/diffusers/models/embeddings/others.py

```diff
@@ -0,0 +1,245 @@
+
+        # replace padding with learnable null embedding
+        xyxy_embedding = xyxy_embedding * masks + (1 - masks) * xyxy_null
+
+        # positionet with text only information
+        if positive_embeddings is not None:
+            # learnable null embedding
+            positive_null = self.null_positive_feature.view(1, 1, -1)
+
+            # replace padding with learnable null embedding
+            positive_embeddings = positive_embeddings * masks + (1 - masks) * positive_null
+
+            objs = self.linears(torch.cat([positive_embeddings, xyxy_embedding], dim=-1))
+
+        # positionet with text and image information
+        else:
+            phrases_masks = phrases_masks.unsqueeze(-1)
+            image_masks = image_masks.unsqueeze(-1)
+
+            # learnable null embedding
+            text_null = self.null_text_feature.view(1, 1, -1)
+            image_null = self.null_image_feature.view(1, 1, -1)
+
+            # replace padding with learnable null embedding
+            phrases_embeddings = phrases_embeddings * phrases_masks + (1 - phrases_masks) * text_null
+            image_embeddings = image_embeddings * image_masks + (1 - image_masks) * image_null
+
+            objs_text = self.linears_text(torch.cat([phrases_embeddings, xyxy_embedding], dim=-1))
+            objs_image = self.linears_image(torch.cat([image_embeddings, xyxy_embedding], dim=-1))
+            objs = torch.cat([objs_text, objs_image], dim=1)
+
+        return objs
+
+
+class LabelEmbedding(nn.Module):
+    """
+    Embeds class labels into vector representations. Also handles label dropout for classifier-free guidance.
+
+    Args:
+        num_classes (`int`): The number of classes.
+        hidden_size (`int`): The size of the vector embeddings.
+        dropout_prob (`float`): The probability of dropping a label.
+    """
+
+    def __init__(self, num_classes, hidden_size, dropout_prob):
+        super().__init__()
+        use_cfg_embedding = dropout_prob > 0
+        self.embedding_table = nn.Embedding(num_classes + use_cfg_embedding, hidden_size)
+        self.num_classes = num_classes
+        self.dropout_prob = dropout_prob
+
+    def token_drop(self, labels, force_drop_ids=None):
+        """
+        Drops labels to enable classifier-free guidance.
+        """
+        if force_drop_ids is None:
+            drop_ids = torch.rand(labels.shape[0], device=labels.device) < self.dropout_prob
+        else:
+            drop_ids = torch.tensor(force_drop_ids == 1)
+        labels = torch.where(drop_ids, self.num_classes, labels)
+        return labels
+
+    def forward(self, labels: torch.LongTensor, force_drop_ids=None):
+        use_dropout = self.dropout_prob > 0
+        if (self.training and use_dropout) or (force_drop_ids is not None):
+            labels = self.token_drop(labels, force_drop_ids)
+        embeddings = self.embedding_table(labels)
+        return embeddings
+
+
+class HunyuanDiTAttentionPool(nn.Module):
```

Comment:
this and the other attention pool are both used by some other embedding((assume it's one of the combined ones), we can move it to where it is used

Reply:
Thanks!

The other `AttentionPool` class is used for `TextTimeEmbedding` (which, in theory, is also a kind of combined embedding class, IMO). 

However `HunyuanDiTAttentionPool` is used in `HunyuanCombinedTimestepTextSizeStyleEmbedding` that combines timesteps, text embeddings, and additional things. So, it's clearly not just text. So, IMO, it's better to keep it in `combined.py`. 

WDYT?

Reply:
>However HunyuanDiTAttentionPool is used in HunyuanCombinedTimestepTextSizeStyleEmbedding that combines timesteps, text embeddings, and additional things. So, it's clearly not just text. So, IMO, it's better to keep it in combined.py

`HunyuanCombinedTimestepTextSizeStyleEmbedding` take a combination if inputs but `HunyuanDiTAttentionPool` is only used to project the text inputs - but I'm fine to put it in `combined.py` since it's probably won't be used on its own. Same with the other `AttentionPool` for `TextTimeEmbedding`

Reply:
Sorry about the iteration here. `TextTimeEmbedding` is really just about projecting `hidden_states`. So, no combination. And also, after taking into consideration what said above, I thought it would make sense to keep `HunyuanCombinedTimestepTextSizeStyleEmbedding` in `image_text`, actually. So, your original suggestion. 

I hope it makes sense now.

Reply:
>So, no combination. And also, after taking into consideration what said above, I thought it would make sense to keep HunyuanCombinedTimestepTextSizeStyleEmbedding in image_text, actually. So, your original suggestion.

sorry when did I suggest to put in `image_text`? it it clearly combined?

Reply:
> HunyuanCombinedTimestepTextSizeStyleEmbedding take a combination if inputs but HunyuanDiTAttentionPool is only used to project the text inputs - but I'm fine to put it in combined.py since it's probably won't be used on its own. Same with the other AttentionPool for TextTimeEmbedding

I thought this meant putting it in "image_text" but you were okay if put in combined as well. Sorry, if I misunderstood it.